lib/dry/validation/messages/resolver.rb

```diff
@@ -90,12 +78,29 @@ def message(rule, tokens: EMPTY_HASH, locale: nil, full: false, path:)
           parsed_tokens = parse_tokens(tokens)
           text = template.(template.data(parsed_tokens))
 
-          [full ? "#{messages.rule(keys.last, msg_opts)} #{text}" : text, meta]
+          [message_text(text, path: path, locale: locale, full: full), meta]
         end
         # rubocop:enable Metrics/AbcSize
 
         private
 
+        def message_text(text, path:, locale: nil, full: false)
+          return text unless full
+
+          key = key_text(path: path, locale: locale)
+
+          [key, text].compact.join(SPACE)
```

Comment:
This should use a similar mechanism like in dry-schema, because since dry-schema 1.5.3, languages that don't use spaces between words are handled properly. See [this PR](https://github.com/dry-rb/dry-schema/pull/292/files) for more details. There's `Dry::Schema::MessageCompiler::FULL_MESSAGE_WHITESPACE` map that you could use here. In the future this should be encapsulated better but for now we can just re-use this constant.

Reply:
Will do that.

Reply:
Should it depend on dry-schema? I've added the same constant to the resolver to not depend on that dry-schema version, but maybe you had something else in mind?